src/transforms/rename_fields.rs

```diff
@@ -0,0 +1,134 @@
+use super::Transform;
+use crate::{
+    event::Event,
+    runtime::TaskExecutor,
+    topology::config::{DataType, TransformConfig, TransformDescription},
+};
+use indexmap::map::IndexMap;
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
+use string_cache::DefaultAtom as Atom;
+use toml::value::Value as TomlValue;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct RenameFieldsConfig {
+    pub fields: IndexMap<String, TomlValue>,
```

Comment:
Maybe, I'm missing something but why can't this just be `IndexMap<Atom, Atom>`?

Reply:
We need to support nested fields and things like `a.b.c = 'd'` come out as `a = { b = { c = 'd' }}}`